src/backend/marsha/core/api/video.py

```diff
@@ -465,6 +478,116 @@ def update_live_state(
 
         return Response({"success": True})
 
+    @action(
+        methods=["get"],
+        detail=True,
+        url_path="pairing-secret",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def pairing_secret(
+        self,
+        request,
+        pk=None,
+    ):
+        """Generate a secret for pairing an external device to a live stream.
+
+        Deletes expired LivePairing objects.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the generated secret.
+        """
+        LivePairing.objects.delete_expired()
+
+        video = self.get_object()
+        if video.live_type != JITSI:
+            return Response(
+                {"detail": "Matching video is not a Jitsi Live."}, status=400
+            )
+
+        try:
+            live_pairing = LivePairing.objects.get(video=video)
+        except LivePairing.DoesNotExist:
+            live_pairing = LivePairing(video=video)
+
+        for _ in range(2):
+            try:
+                live_pairing.generate_secret()
+                live_pairing.save()
+                break
+            except ValidationError:
+                pass
+
+        serializer = serializers.LivePairingSerializer(instance=live_pairing)
+        return Response(serializer.data)
+
+
+class PairingChallengeThrottle(AnonRateThrottle):
+    """Throttling for pairing challenge requests."""
+
+    def get_ident(self, request):
+        """Identify requests by box_id instead of IP"""
+        try:
+            box_id = request.data.get("box_id")
+            Device.objects.get(pk=box_id)
+            return box_id
+        except (Device.DoesNotExist, ValidationError):
```

Comment:
Why `ValidationError` ?

Reply:
Because sending a wrong box_id (that is not an uuid), like in `test_api_video_pairing_challenge_wrong_box_id`, would raise it.

Reply:
:+1: